### ISSUE TITLE
Option to exclude from pruning based on number of pulls

### ISSUE THREAD

**Issue by cicirello** [vlaurin/action-ghcr-prune#35]
It would be nice if there was an option to exclude containers from pruning if they were pulled more than some threshold. Perhaps controlled by an integer input `skip-if-pulled-more-than` or something similar.

**Comment by vlaurin** [comment]
I totally agree that it would be a good feature to have, I just had a look at it.

Unfortunately, GitHub doesn't expose that information in their API as far as I can see (cf. [List package versions for a package owned by an organization](https://docs.github.com/en/rest/packages#list-package-versions-for-a-package-owned-by-an-organization)). I checked by calling the API directly and the documentation is accurate, there's nothing relating to number of pulls in there.

So as it stands and until such time this becomes exposed by the GitHub API, sadly this is not an option I can add to the action.

**Comment by cicirello** [comment]
@vlaurin the `PackageVersion` object has a field `statistics` which is a [`PackageVersionStatistics`](https://docs.github.com/en/graphql/reference/objects#packageversionstatistics) that contains the count.

**Comment by vlaurin** [comment]
Thank you for pointing that out, I'll have a look at it.

**Comment by davidboweninrupt** [comment]
I'm hoping to use this feature to delete all my old images with zero or one pulls.
